background.js

```diff
@@ -46,83 +45,88 @@ chrome.runtime.onMessage.addListener(
-		initStorage(rcxDict.prototype.numList[i*2], "true")
-	}
+			// v0.8.6
+			initStorage("onlyreading", false);
+			// v0.8.8
+			if (items.highlight == "yes")
+				items.highlight = "true";
+			if (items.highlight == "no")
+				items.highlight = "false";
+			if (items.textboxhl == "yes")
+				items.textboxhl = "true";
+			if (items.textboxhl == "no")
+				items.textboxhl = "false";
+			if (items.onlyreading == "yes")
+				items.onlyreading = "true";
+			if (items.onlyreading == "no")
+				items.onlyreading = "false";
+			initStorage("copySeparator", "tab");
+			initStorage("maxClipCopyEntries", "7");
+			initStorage("lineEnding", "n");
+			initStorage("minihelp", "true");
+			initStorage("disablekeys", "false");
+			initStorage("kanjicomponents", "true");
 
-    // v0.8.92
-	initStorage("popupDelay", "150");
-	initStorage("showOnKey", "");
-}
+			for (i = 0; i * 2 < rcxDict.prototype.kanjiInfoLabelList.length; i++) {
+				initStorage(rcxDict.prototype.kanjiInfoLabelList[i * 2], "true")
+			}
 
-/** 
-* Initializes the localStorage for the given key. 
-* If the given key is already initialized, nothing happens. 
-* 
-* @author Teo (GD API Guru)
-* @param key The key for which to initialize 
-* @param initialValue Initial value of localStorage on the given key 
-* @return true if a value is assigned or false if nothing happens 
-*/ 
-function initStorage(key, initialValue) {
-  var currentValue = localStorage[key];
-  if (!currentValue) {
-	localStorage[key] = initialValue;
-	return true;
-  }
-  return false;
-}
+			// v0.8.92
+			initStorage("popupDelay", "150");
+			initStorage("showOnKey", "");
+		}
 
-rcxMain.config = {};
-rcxMain.config.css = localStorage["popupcolor"];
-rcxMain.config.highlight = localStorage["highlight"];
-rcxMain.config.textboxhl = localStorage["textboxhl"];
-rcxMain.config.onlyreading = localStorage["onlyreading"];
-rcxMain.config.copySeparator = localStorage["copySeparator"];
-rcxMain.config.maxClipCopyEntries = localStorage["maxClipCopyEntries"];
-rcxMain.config.lineEnding = localStorage["lineEnding"];
-rcxMain.config.minihelp = localStorage["minihelp"];
-rcxMain.config.popupDelay = parseInt(localStorage["popupDelay"]);
-rcxMain.config.disablekeys = localStorage["disablekeys"];
-rcxMain.config.showOnKey = localStorage["showOnKey"];
-rcxMain.config.kanjicomponents = localStorage["kanjicomponents"];
-rcxMain.config.kanjiinfo = new Array(rcxDict.prototype.numList.length/2);
-for (i = 0; i*2 < rcxDict.prototype.numList.length; i++) {
-	rcxMain.config.kanjiinfo[i] = localStorage[rcxDict.prototype.numList[i*2]];
-}
+		/**
+		 * Initializes the localStorage for the given key.
+		 * If the given key is already initialized, nothing happens.
+		 *
+		 * @author Teo (GD API Guru)
+		 * @param key The key for which to initialize
+		 * @param initialValue Initial value of localStorage on the given key
+		 * @return boolean if a value is assigned or false if nothing happens
+		 */
+		function initStorage(key, initialValue) {

```

Comment:
You've moved initStorage inside of this callback which is a bit weird. I guess it's only used inside the callback but it does look a bit weird to have the function defined here. I would move it to global scope for consistency.

Reply:
done.

> You've moved initStorage inside of this callback which is a bit weird. I guess it's only used inside the callback but it does look a bit weird to have the function defined here. I would move it to global scope for consistency.

Done...